src/core/cardano/walletConnect/peerConnection.ts

```diff
@@ -80,10 +82,20 @@ class PeerConnection {
       this.eventService
     );
     this.identityWalletConnect.setOnConnect(
-      (connectMessage: IConnectMessage) => {
+      async (connectMessage: IConnectMessage) => {
         if (!connectMessage.error) {
           this.connected = true;
-          this.connectedDAppAdress = connectMessage.dApp.address;
+          const { name, url, address } = connectMessage.dApp;
+          this.connectedDAppAdress = address;
+          await Agent.agent.peerConnectionMetadataStorage.updatePeerConnectionMetadata(
+            address,
+            {
+              name,
+              selectedAid,
+              url,
+              iconB64: ICON_BASE64,
```

Comment:
There's now a `1.2.17` version of peer connect which passes `icon` in the connectMessage.

I think we can do `icon || ICON_BASE64`

Maybe also make sure that the icon is base64 if it's passed, and if not, also call back to `ICON_BASE64`